Automattic/wp-calypso
# Domain step test: Change filter icon

#### Changes proposed in this Pull Request
The next domain step test is being implemented in #39276. The test is hidden behind a feature flag. We will build individual pieces of the UI in separate PRs. 

**This PR**:
* Removes the Filter icon from the search input box and adds a toggle for exact match.
* The first version included a toggle([screenshot-1](https://user-images.githubusercontent.com/1269602/74146885-17356700-4c28-11ea-8ef9-f141f4331bb3.png) and [screenshot-2](https://user-images.githubusercontent.com/1269602/74146905-1ef50b80-4c28-11ea-8eaa-62f980a84510.png)) for exact match, and upon design iteration we decided to remove the toggle.

**Without search term**

<img width="1056" alt="Screenshot 2020-02-13 at 12 24 17 PM" src="https://user-images.githubusercontent.com/1269602/74409654-323cec80-4e5d-11ea-8680-3bda3cfea74c.png">

**With search term**

<img width="1020" alt="Screenshot 2020-02-13 at 12 29 00 PM" src="https://user-images.githubusercontent.com/1269602/74409304-77acea00-4e5c-11ea-9f22-42a082dad9d1.png">

**With filter applied(filter icon highlighted)**

<img width="1034" alt="Screenshot 2020-02-13 at 12 29 09 PM" src="https://user-images.githubusercontent.com/1269602/74409335-82677f00-4e5c-11ea-8999-e3abeae8ec2f.png">

**Mobile screen size**

![hash-bbdb61f3319c671ef38028752cebf3204e9a6dd9 calypso live_start_ecommerce-onboarding_domains(iPhone X)](https://user-images.githubusercontent.com/1269602/74409410-a925b580-4e5c-11ea-9897-42647994e301.png)


#### Testing instructions


Note: The popover on clicking the filter icon will have extra filter options("Max characters" and "Enable dashes") in "dev" environment. We don't yet want these extra filters in production, so test using the calypso.live link to not see these extra options.

* Begin a fresh signup at /start or click "Add new site". You need to be in the _variantShowUpdates_  variant of **domainStepCopyUpdates**  test and _variantDesignUpdates_ variant of **domainStepDesignUpdates**.
* At the domain step, verify the following:
   - Search with no filter selected, and verify that the search results contain terms that are not exact matches
    - In the input box, before typing anything first apply the exact match filter. Then start typing and verify that the search results contain terms that are only exact matches
    - With the input box having a search term and search results displayed, apply/remove the exact match filter and verify that the search results update.
* Verify that UI matches the version shown in [this comment](https://github.com/Automattic/wp-calypso/pull/39334#issuecomment-584893982).
* Select any domain and go to the Plans step, then click the Back button to go back to the Domain step. Verify that the earlier toggle state is remembered.

**Scenario 2**

* Assign yourself to the "control" group of domainStepCopyUpdates test and begin signup at /start.
* Verify that Automattic/wp-calypso#39337 and Automattic/wp-calypso#39338 are fixed.

**Scenario 3**

* Begin a fresh signup at /start or click "Add new site". Assign yourself to the _variantShowUpdates_  variant of **domainStepCopyUpdates**  test and _control_ group of **domainStepDesignUpdates**.
* Verify that you do not see the changes implemented in this PR - you should see the earlier gear icon for filter.
